### PR TITLE
Clarify user experience flow raised in issue #114

### LINKED ISSUE #114: Figure 3 is too hard to understand
Figure 3, and its discussion is really hard to follow and I think that figure needs a rewrite.

Some issues include:

- There's an "App Store" role in the header of the figure that isn't used, but should be (in 2a).
- Step 3 is illustrated as installing the Untrusted App from the TAM, which contradicts the text saying it's installed from the App Store.
- Text discussion says "The Untrusted Application will get messages from TAM" which is not true... the Broker does.
- The certs shown at the top of the diagram are not mentioned in the text discussion.
- I can't parse the sentence above the figure ("A device administrator may choose to accept a subset of the allowed TAs via consent or action of downloading.")

### PR DESCRIPTION
**Pull request by mingpeiwk**
fix #114

### CODE REVIEW

**Comment by mingpeiwk** [comment]
I added an abbreviation for "Untrusted Application" for figure 3. I use "UTA" to compare with "TA". Is this good for you? Thanks.

File: draft-ietf-teep-architecture.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ TEE. Typically, application components are chosen to execute inside a TEE becaus
 those application components perform security sensitive operations or operate on
 sensitive data. An application component running inside a TEE is referred to as a
 Trusted Application (TA), while an application running outside any TEE
-is referred to as an Untrusted Application.
+is referred to as an Untrusted Application (UTA).
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md]: I do not think we should introduce an acronym here.  (We already got rid of "UA" as an acronym in earlier PRs.)

File: draft-ietf-teep-architecture.md [hunk 2]
```diff
@@ -619,33 +619,30 @@ complex than Cases 2 and 3.
 ## Entity Relations
 
 This architecture leverages asymmetric cryptography to
-authenticate a device to a TAM. Additionally, a TEE
-in a device authenticates a TAM and TA signer. The
+authenticate a device to a TAM. Additionally, a TEE Agent
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md]: "TEEP Agent", not "TEE Agent"

File: draft-ietf-teep-architecture.md [hunk 3]
```diff
@@ -619,33 +619,30 @@ complex than Cases 2 and 3.
...
-       TA ----------------- 2b. Supply ------> | 4. Messaging-->|
-        |                           |          |                |
+     UTA, TA
+        |               |           |                |            |
+        |               |           |                |            |
+       UTA -- 2a. --->  | --- 3. Install ----------> |            |
+       TA --- 2b. Supply ---------> | 4. Messaging-->|            |
+        |               |           |                |            |
 ~~~~
 {: #experience title="Developer Experience"}
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md:649]: Line 653-654: The same developer building both the Untrusted App and TA is just one example.
So "shows an example where the same developer builds two applications:"

(Above also fixes the typo of "an TA")

File: draft-ietf-teep-architecture.md [hunk 4]
```diff
@@ -664,8 +661,12 @@ that will be managing the TA in various devices. At step 3, a user
 will go to an Application Store to download the Untrusted
 Application. The Untrusted Application will trigger TA installation
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md]: "The Untrusted Application will trigger" is possible, but I wouldn't say it's a desirable developer experience.
Recommend "Since the Untrusted Application depends on the TA, installing the Untrusted Application will trigger" (which tries to leave it ambiguous whether it's the app or the installer that does the actual trigger).

File: draft-ietf-teep-architecture.md [hunk 5]
```diff
@@ -664,8 +661,12 @@ that will be managing the TA in various devices. At step 3, a user
...
 by initiating communication with a TAM. This is the step 4. The Untrusted Application
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md]: grammar: "the step 4" -> "step 4"

File: draft-ietf-teep-architecture.md [hunk 6]
```diff
@@ -664,8 +661,12 @@ that will be managing the TA in various devices. At step 3, a user
...
-will get messages from TAM, and interacts with device
-TEE via an Agent.
+will interact with TAM via a TEEP Broker that faciliates communications between a TAM
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md:666]: I don't think it's ever correct to say that the "Untrusted Application will interact with a TAM".
Change "Untrusted Application will" to "TEEP Agent will".

File: draft-ietf-teep-architecture.md [hunk 7]
```diff
@@ -664,8 +661,12 @@ that will be managing the TA in various devices. At step 3, a user
...
+and the TEEP Agent in TEE.
+
+Some TA installation might ask for a user's consent. On the other hand,
+a Device Administrator may choose what Untrusted Applications and related TAs to
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md]: grammar: "may" -> "might"

File: draft-ietf-teep-architecture.md [hunk 8]
```diff
@@ -664,8 +661,12 @@ that will be managing the TA in various devices. At step 3, a user
...
+be installed. A user consention flow is out of scope of the TEEP architecture.
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md]: grammar: "consention" -> "consent"

**Review by dthaler — CHANGES_REQUESTED** [review]

**Comment by dthaler** [comment]
@mingpeiwk, I pushed changes to this PR to address my feedback, please review.

**Review by dthaler — APPROVED** [review]
Ok with me (with my edits)

**Comment by dthaler** [comment]
Merging so it can get into draft -06
